ironfish/src/consensus/verifier.ts

```diff
@@ -209,6 +210,12 @@ export class Verifier {
       return verificationResult
     }
 
+    // Currently we only support one transaction version. This is checked when
+    // we call workerPool.verify but doing it here as well for efficiency
+    if (transaction.version() !== TRANSACTION_VERSION) {
+      return { valid: false, reason: VerificationResultReason.INVALID_TRANSACTION_VERSION }
+    }
```

Comment:
This is probably not that performance-intensive, but this could be checked in `verifyBlock` as well before calling `verifyTransactions` on a set of transactions

Reply:
I think I might skip on that one. Initially sounded like a good idea but was thinking about it a little more. I just added the call here for performance reasons since this is the code path for checking gossiped transactions. If a transaction is gossiped to us we want to validate any errors as efficiently as possible to avoid DDoS. If DDoS wasn't an issue I don't think I'd add it here. With blocks though we have a built in DDoS protection in that someone has to mine the block and put in that energy if they want to submit an invalid block. So I don't think we have to worry about DDoS when checking transactions in blocks